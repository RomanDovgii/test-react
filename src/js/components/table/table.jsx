import React, {PureComponent} from "react";
import Pagination from "../pagination/pagination";
import TableHeading from "../table-heading/table-heading";
import TableElement from "../table-element/table-element";
import {connect} from "react-redux";
import {tableType} from "../../types/types";
import {fetchData} from "../../store/action/api-action";

class Table extends PureComponent {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    const {loadData, currentPageUsers} = this.props;

    if (currentPageUsers.length === 0) {
      loadData();
    }
  }

  render() {
    const {currentPageUsers} = this.props;
    return (
      <section className="table">
        <table className="table__table">
          <thead>
            <TableHeading/>
          </thead>
          <tbody>
            {
              currentPageUsers.map((user) => {
                return (
                  <TableElement
                    key = {`${user.id}+${user.lastName}+${user.firstNmae}`}
                    user = {user}
                  />
                );
              })
            }
          </tbody>
        </table>
        <Pagination/>
      </section>
    );
  }
}

Table.propTypes = tableType;

const mapDispatchToPorps = (dispatch) => ({
  loadData() {
    dispatch(fetchData(`LARGE`));
  }
});

const mapStateToProps = (state) => ({
  currentPageUsers: state.currentPageUsers
});

export {Table};
export default connect(mapStateToProps, mapDispatchToPorps)(Table);
