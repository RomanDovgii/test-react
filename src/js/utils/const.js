export const MAXIMUM_USERS_PER_PAGE = 50;

export const AppRoute = {
  SELECT: `/select`,
  TABLE: `/table`
};
