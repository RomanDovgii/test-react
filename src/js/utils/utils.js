export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const sortToTop = (a, b) => {
  switch (true) {
    case a > b:
      return -1;
    case a < b:
      return 1;
    default:
      return 0;
  }
};

export const sortToBot = (a, b) => {
  switch (true) {
    case a < b:
      return -1;
    case a > b:
      return 1;
    default:
      return 0;
  }
};
