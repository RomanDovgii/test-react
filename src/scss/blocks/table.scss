.table {
  display: flex;
  flex-direction: column;
  margin-top: 30px;
}

.table__table {
    border-collapse: collapse;
    width: 100%;
    max-width: 1340px;
}

.table__heading, .table__data {
    border: 2px solid #333333;
}

.table__data {
    text-align: center;

    padding: 10px 10px 10px 10px;
}

.table__row {
    cursor: pointer;
}

.table__button {
  border: none;
  background: none;

  position: relative;
  box-sizing: border-box;
  top: 0;
  left: 0;
  min-width: 90px;
  width: 100%;
  height: 100%;
  font-weight: 700;
  cursor: pointer;

  padding: 10px 10px 10px 10px;

  &--top-to-bot {
    &::after {
        content: "";
        box-sizing: border-box;

        display: block;
        width: 5px;
        height: 10px;

        position: absolute;
        top: 50%;
        right: 10px;

        transform: translateY(-5px);

        border-top: 10px solid black;
        border-left: 5px solid transparent;
        border-right: 5px solid transparent;
    }
  }

  &--bot-to-top {
    &::after {
        content: "";
        box-sizing: border-box;

        display: block;
        width: 5px;
        height: 10px;

        position: absolute;
        top: 50%;
        right: 10px;

        transform: translateY(-5px);

        border-bottom: 10px solid black;
        border-left: 5px solid transparent;
        border-right: 5px solid transparent;
    }
  }
}
