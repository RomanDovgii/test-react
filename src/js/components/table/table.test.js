import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import {Table} from "./table";
import {noop} from "../../utils/utils";

const mockUser = {
  id: 339,
  firstName: `Susen`,
  lastName: `Graham`,
  email: `LNicholson@mi.org`,
  phone: `(220)920-9427`,
  address: {
    streetAddress: `7422 Vitae Rd`,
    city: `Bergen County`,
    state: `NE`,
    zip: `24099`
  },
  description: `aenean augue neque mattis sollicitudin turpis sed aenean elit fringilla nec massa libero nullam lectus aenean et vel sed pulvinar et augue facilisis vitae libero dolor massa adipiscing dolor vitae orci tortor`
};

it(
    `Should render Table`,
    () => {
      const renderer = new ShallowRenderer();

      const tree = renderer
      .render(
          <Table
            isLoading={false}
            loadData={noop}
            currentPageUsers={[mockUser, mockUser, mockUser]}
          />
      );

      expect(tree).toMatchSnapshot();
    }
);

it(
    `Should render loading Table`,
    () => {
      const renderer = new ShallowRenderer();

      const tree = renderer
      .render(
          <Table
            isLoading={true}
            loadData={noop}
            currentPageUsers={[]}
          />
      );

      expect(tree).toMatchSnapshot();
    }
);
