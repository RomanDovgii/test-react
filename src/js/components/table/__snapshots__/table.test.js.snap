// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Table 1`] = `
<section
  className="table"
>
  <table
    className="table__table"
  >
    <thead>
      <Memo(Connect(TableHeading)) />
    </thead>
    <tbody>
      <Memo(Connect(TableElement))
        user={
          Object {
            "address": Object {
              "city": "Bergen County",
              "state": "NE",
              "streetAddress": "7422 Vitae Rd",
              "zip": "24099",
            },
            "description": "aenean augue neque mattis sollicitudin turpis sed aenean elit fringilla nec massa libero nullam lectus aenean et vel sed pulvinar et augue facilisis vitae libero dolor massa adipiscing dolor vitae orci tortor",
            "email": "LNicholson@mi.org",
            "firstName": "Susen",
            "id": 339,
            "lastName": "Graham",
            "phone": "(220)920-9427",
          }
        }
      />
      <Memo(Connect(TableElement))
        user={
          Object {
            "address": Object {
              "city": "Bergen County",
              "state": "NE",
              "streetAddress": "7422 Vitae Rd",
              "zip": "24099",
            },
            "description": "aenean augue neque mattis sollicitudin turpis sed aenean elit fringilla nec massa libero nullam lectus aenean et vel sed pulvinar et augue facilisis vitae libero dolor massa adipiscing dolor vitae orci tortor",
            "email": "LNicholson@mi.org",
            "firstName": "Susen",
            "id": 339,
            "lastName": "Graham",
            "phone": "(220)920-9427",
          }
        }
      />
      <Memo(Connect(TableElement))
        user={
          Object {
            "address": Object {
              "city": "Bergen County",
              "state": "NE",
              "streetAddress": "7422 Vitae Rd",
              "zip": "24099",
            },
            "description": "aenean augue neque mattis sollicitudin turpis sed aenean elit fringilla nec massa libero nullam lectus aenean et vel sed pulvinar et augue facilisis vitae libero dolor massa adipiscing dolor vitae orci tortor",
            "email": "LNicholson@mi.org",
            "firstName": "Susen",
            "id": 339,
            "lastName": "Graham",
            "phone": "(220)920-9427",
          }
        }
      />
    </tbody>
  </table>
  <Memo(Connect(Pagination)) />
</section>
`;

exports[`Should render loading Table 1`] = `<Loading />`;
